pytorch_lightning/loops/closure.py

```diff
@@ -53,25 +54,29 @@ def __init__(self) -> None:
         super().__init__()
         self._result: Optional[ClosureResult] = None
 
-    def get_result(self) -> Optional[ClosureResult]:
+    def consume_result(self) -> ClosureResult:
         """The cached result from the last time the closure was called.
 
         Once accessed, the internal reference gets reset and the consumer will have to hold on to the reference as long
         as necessary.
         """
-        result = self._result
-        self._result = None  # free memory
+        if self._result is None:
+            raise MisconfigurationException(
```

Comment:
@carmocca FB had a problem here where they got the error message but not calling the closure was actually intentional. Are we ok with downgrading this to a warning message?

Reply:
We can downgrade this to a warning if necessary, however, if there isn't a reason for allowing this, I still believe this should be an error.

We can make this a warning and a deprecation and then convert this back to an error two minor releases later.

@ananthsub what do you think of this solution? Do you have an argument for why skipping the closure should be allowed?